### PR TITLE
Add environment variables needed for Spark2 and Kafka

### PR DESCRIPTION
**Pull request by kirtiv1**
…ations

### CODE REVIEW

File: start.py [hunk 1]
```diff
@@ -721,10 +721,10 @@ def _configure_sdc(deployment, cluster, is_kerberos_enabled):
                                                   'type': 'STREAMSETS',
                                                   'displayName': 'StreamSets',
                                                   'roles': [datacollector_role]}])
-    # When running an application on YARN, the Spark executor requires access to the spark-submit script located in
-    # the Spark installation directory. Default is directory specified by SPARK_HOME environment variable.
-    # Hence SPARK_HOME environment variable must be set before starting StreamSets Data Collector.
-    configs = {'sdc-env.sh_role_safety_valve': 'export SPARK_HOME=/opt/cloudera/parcels/CDH/lib/spark'}
+    # When running an application  with Spark2, the following
+    # environment variables must be set before starting StreamSets Data Collector.
+    config_value = 'export SPARK_SUBMIT_YARN_COMMAND=/usr/bin/spark2-submit\nexport SPARK_KAFKA_VERSION=0.10'
```
Comment (dimaspivak) [inline start.py]: How about:
```
environment_variables = {'SPARK_SUBMIT_YARN_COMMAND': '/usr/bin/spark2-submit',                                                   
                         'SPARK_KAFKA_VERSION': '0.10'}                                                                           
configs = {'sdc-env.sh_role_safety_valve': '\n'.join('export {}={}'.format(key, value)                                            
                                                     for key, value in environment_variables.items())}
```

Comment (dimaspivak) [inline start.py]: Is it really cool like this to hardcode this `SPARK_KAFKA_VERSION` thing or will it potentially bite us at some point?

**Review by dimaspivak — CHANGES_REQUESTED** [review]

File: start.py (hunk 1 above)
Comment (ujunko) [reply start.py]: SPARK_KAFKA_VERSION is a version of Spark x Kafka integration, not just Kafka version. There are only two versions now, Spark2 x Kafka 0.9- and Spark2 x Kafka0.10+. We are testing against later one. Even if user uses Kafka 0.11 or 0.12+, we will keep using "SPARK_KAFKA_VERSION=0.10". I think it makes sense to hardcode this way.

Comment (kirtiv1) [reply start.py]: Done.

**Review by dimaspivak — APPROVED** [review]

**Pull request merged**